### PR TITLE
[aptos-vm] re-expose multi-agent transactions

### PR DESCRIPTION
**Pull request by davidiw**
* add an example for direct transfers for nfts
* demonstrate that a smoke test

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/Account.move [hunk 1]
```diff
@@ -263,17 +264,38 @@ module AptosFramework::Account {
...
-        _txn_max_gas_units: u64,
-        _txn_expiration_time: u64,
-        _chain_id: u8,
-    ) {
-        assert!(false, Errors::invalid_argument(EMULTI_AGENT_NOT_SUPPORTED));
+        sender: signer,
+        txn_sequence_number: u64,
+        txn_sender_public_key: vector<u8>,
+        secondary_signer_addresses: vector<address>,
+        secondary_signer_public_key_hashes: vector<vector<u8>>,
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/Account.move:271]: it's very interesting that we do the hash inside move for primary but not for secondary lol

**Review by zekun000 — APPROVED** [review]
amazing! so where's all the deleted multi-agent tests?

File: testsuite/smoke-test/src/nft_transaction.rs [hunk 2]
```diff
@@ -100,6 +100,13 @@ impl AptosTest for NFTTransaction {
         let claim_txn = owner.sign_with_transaction_builder(claim_builder);
         client.submit_and_wait(&claim_txn).await?;
 
+        let transfer_builder = ctx.transaction_factory().payload(
+            aptos_stdlib::encode_direct_transfer_script_function(creator.address(), token_num, 1),
+        );
+        let transfer_txn =
+            owner.sign_multi_agent_with_transaction_builder(vec![&creator], transfer_builder);
+        client.submit_and_wait(&transfer_txn).await?;
+
```
Comment (davidiw) [inline testsuite/smoke-test/src/nft_transaction.rs:109]: all of them are here

hahaha
all those were hacks, I'm not going to try to bring them back.

File: aptos-move/framework/aptos-framework/sources/Account.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/Account.move:271]: yes... I hadn't noticed and then had to double back to fix my code 😭 

we should probably unify this awkwardness...

I also wonder if there is some way we can offer this cool function nonsense to other apps.

**Pull request merged**